flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -708,7 +708,7 @@ public ContextResolvedTable getTableOrError(ObjectIdentifier objectIdentifier) {
         return getTable(objectIdentifier)
                 .orElseThrow(
                         () ->
-                                new TableException(
+                                new ValidationException(
```

Comment:
I would associate a validation error as something incorrect or inconsistent with the definition. For me attempting to get a table that does not exist is not a validation error or something that occurs in the validation phase. What do others think?

Reply:
a comment about `TableException` https://github.com/apache/flink/blob/d08abb7b1697a3ba78ee0bc5d31f841eb0c9d386/flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableException.java#L24-L29

table not exists is a user error since a user supplies a table name to get. If a user supplied a name for not existed table then it is a user error